railgun/internal/proxy/clientgo_cached_proxy_resolver.go

```diff
@@ -295,13 +300,23 @@ func (p *clientgoCachedProxyResolver) updateKongAdmin() error {
 	// generate the deck configuration to be applied to the admin API
 	targetConfig := deckgen.ToDeckContent(p.ctx, p.deprecatedLogger, kongstate, nil, nil)
 
+	// retrieve custom entities
+	var customEntities []byte
+	if p.kongConfig.InMemory && p.KongCustomEntitiesSecret != "" {
+		customEntities, err = fetchCustomEntities(p.KongCustomEntitiesSecret, storer)
+		if err != nil {
+			// failure to fetch custom entities shouldn't block updates
```

Comment:
The error here would generally be a cache miss right? Should we differentiate between errors and potentially fail unless we're sure it's just a cache miss?

Reply:
AFAIK the store will fill cache if it misses; the failure here happens because of misconfiguration rather than a transient issue. Whether this should be a hard failure or not is a UX question, but the soft failure here is for parity with the existing 1.x behavior: https://github.com/Kong/kubernetes-ingress-controller/blob/v1.2.0/internal/ingress/controller/kong.go#L42-L45